packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -61,6 +61,10 @@ export class SCWSigner implements Signer {
   }
 
   async handshake(args: RequestArguments) {
+    // Open the popup before constructing the request message.
+    // This is to ensure that the popup is not blocked by some browsers (i.e. Safari)
+    await this.communicator.waitForPopupLoaded?.();
+
```

Comment:
how does this impact the UX. Do we see longer loading times on the popup for the key exchange?

Reply:
theoretically yea, but in my local testing, i couldn't notice the difference. this pattern is already being used for regular request, where it requires message encryption and also early loaded the popup to bypass popup blocking, at [here](https://github.com/coinbase/coinbase-wallet-sdk/blob/felix/coinbase_handshake/packages/wallet-sdk/src/sign/scw/SCWSigner.ts#L147C1-L150C52)

Reply:
Does this impact `eth_requestAccounts` at all?

Reply:
theoretically yes, for `eth_requestAccounts`, by doing this, we are loading popup a bit sooner. 

the time delta would be considered very small

Before
1. User clicking the button
2. SDK prepping handshake message envelope
3. SDK opens popup
4. SDK pass envelope to popup

The time delta between 1 and 3 is the time used for prepping message envelope, this time is considered very small, that it does not trigger mobile browser popup blocking policy.

After
1. User clicking the button
2. SDK opens popup
3. SDK prepping handshake message envelope
4. SDK pass envelope to popup

so I think moving 3 to be between popup opening and passing envelope to popup won't have a big impact on UX